trainings/insights/lesson-plan.md

```diff
@@ -0,0 +1,83 @@
+
+
+# Code Insights Webinar
+
+**Topic:** This training is intended as an initial deep-dive to educate users on Code Insights outside of the 101/102 context. **If at all possible, we recommend having the customer provide a Code Insights use case to you *before* the training, giving you the opportunity to work on that specific insight or something similar;** if not, you can use this as a way to show Code Insights to them and discuss use cases they might find valuable live on the training.
+
+## Intro
+
+Today's training is intended to walk users through our Code Insights product, and show users how to use the feature. By the end of it, we hop that you'll feel comfortable with what Code Insights is and can do, and will be ready to start with your own changes in Sourcegraph.
+
+❗️ If the training is happening in the context of a champion brining in other team members, have them speak at the top about why they think Code Insights will be useful, and what they envision the team using it for.
+
+## Unit 1: Get familiar with Code Insights
+
+**Learning goals:** After completing this unit, customers will understand what Code Insights are, and can be used to track.
+
+### What is Code Insights?
+
+Code Code Insights reveals high-level information about your codebase, based on both how your code changes over time and its current state. It's a way of visualizing your searches as graphs, so that you can track not just what things are like today, but how that has changed over time. This can be super helpful for tracking things like:
+
+* Migrating from one version of a language or dependency to another
+* Whether folks are adding frequent linter overrides 
+* Tracking how many TODOs are present in your code
+* How adoption of a new API is going
+
+With this info, you can easily make informed decisions about priorities and where to focus your time and energy.
+
+### Running an Insight
+
+To get started, I'm going to generate an insight that tracks TypeScript utilization versus JavaScript utilization in my codebase.
+
+❗️ This is a generic starting point. If you know a different Insight will be more appropriate, or the customer has a specific use case in mind, you may want to walk through that change, instead. A full list of recipes is available [in the docs](https://docs.sourcegraph.com/code_insights/references/common_use_cases). 
+
+🔎 The trainer should demonstrate:
+
+* Creating an insight that tracks [JavaScript vs. TypeScript](https://docs.sourcegraph.com/code_insights/references/common_use_cases#language-use-over-time)
+* Review what the `lang:` and `select:` filters are doing
+* Show that once the insight is generated, you can filter by repo
+* Show how to access individual data points by clicking in to them
+* Show how to get a shareable link for the insight
```

Comment:
Worth mentioning at the end, you can do all of this very quickly via a diff search + hitting create insight